statsmodels/graphics/functional.py

```diff
@@ -1,14 +1,441 @@
+        created.
+
+    Returns
+    -------
+    fig : Matplotlib figure instance
+        If `ax` is None, the created figure.  Otherwise the figure to which
+        `ax` is connected.
+    hdr_res : HdrResults instance
+        An `HdrResults` instance with the following attributes:
+
+         - 'median', array. Median curve.
+         - 'hdr_50', array. 50% quantile band. [sup, inf] curves
+         - 'hdr_90', list of array. 90% quantile band. [sup, inf]
+            curves.
+         - 'extra_quantiles', list of array. Extra quantile band.
+            [sup, inf] curves.
+         - 'outliers', ndarray. Outlier curves.
+
+    See Also
+    --------
+    banddepth, rainbowplot, fboxplot
+
+    Notes
+    -----
+    The median curve is the curve with the highest probability on the reduced
+    space of a Principal Component Analysis (PCA).
+
+    Outliers are defined as curves that fall outside the band corresponding
+    to the quantile given by `threshold`.
+
+    The non-outlying region is defined as the band made up of all the
+    non-outlying curves.
+
+    Behind the scene, the dataset is represented as a matrix. Each line
+    corresponding to a 1D curve. This matrix is then decomposed using Principal
+    Components Analysis (PCA). This allows to represent the data using a finite
+    number of modes, or components. This compression process allows to turn the
+    functional representation into a scalar representation of the matrix. In
+    other words, you can visualize each curve from its components. Each curve
+    is thus a point in this reduced space. With 2 components, this is called a
+    bivariate plot (2D plot).
+
+    In this plot, if some points are adjacent (similar components), it means
+    that back in the original space, the curves are similar. Then, finding the
+    median curve means finding the higher density region (HDR) in the reduced
+    space. Moreover, the more you get away from this HDR, the more the curve is
+    unlikely to be similar to the other curves.
+
+    Using a kernel smoothing technique, the probability density function (PDF)
+    of the multivariate space can be recover. From this PDF, it is possible to
+    compute the density probability linked to the cluster of points and plot
+    its contours.
+
+    Finally, using these contours, the different quantiles can be extracted
+    allong with the median curve and the outliers.
+
+    References
+    ----------
+    [1] R.J. Hyndman and H.L. Shang, "Rainbow Plots, Bagplots, and Boxplots for
+        Functional Data", vol. 19, pp. 29-45, 2010.
+
+    Examples
+    --------
+    Load the El Nino dataset.  Consists of 60 years worth of Pacific Ocean sea
+    surface temperature data.
+
+    >>> import matplotlib.pyplot as plt
+    >>> import statsmodels.api as sm
+    >>> data = sm.datasets.elnino.load()
+
+    Create a functional boxplot.  We see that the years 1982-83 and 1997-98 are
```

Comment:
the legend show 1951-1953 as outlier labels when I run the testcase example.
there might be a bug

Reply:
Fixed, only the plots were affected. Was due to a difference between the output of np.all(np.isin()) and np.where(). (Indices instead of full list of booleans)